tests/system/action/motion/test_create_sequence_number.py

```diff
@@ -0,0 +1,256 @@
+            "/",
+            json=[
+                {
+                    "action": "motion.create",
+                    "data": [
+                        {
+                            "title": "motion_title",
+                            "meeting_id": 222,
+                            "workflow_id": 12,
+                            "text": "test2",
+                        }
+                    ],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        model = self.get_model("motion/2")
+        self.assertEqual(model.get("sequence_number"), 2)
+
+    def test_create_sequence_numbers_race_condition(self) -> None:
+        """
+        The main thread in this testing function creates an instance of a threading.lock and acquires it.
+        Thread1 will be started first, but should wait for the acquired lock after reading and before writing.
+        Thread2 will be started later, but passes thread1, because do not acquire the lock.
+        After finishing thread2 the lock will be released in main thread and thread1 could continue with it's writing part.
+
+        The lock-object will be shared in in threading.local(), instance created in shared/lock.py.
+        If possible you should pass as an argument to the thread function(s).
+        """
+        pytest_thread_local.name = "MainThread_RC"
+        self.create_model("meeting/222", {"name": "meeting222"})
+        self.create_workflow(workflow_id=12, meeting_id=222)
+        self.create_workflow(workflow_id=13, meeting_id=222)
+
+        # with MonkeyPatch_DatastoreAdapter_write():
+        with monkeyPatch_DatastoreAdapter_write():
+            testlock = threading.Lock()
+            thread1 = threading.Thread(
+                target=thread_method,
+                kwargs={
+                    "test_instance": self,
+                    "motion_title": "Interrupted motion",
+                    "meeting_id": 222,
+                    "workflow_id": 12,
+                    "testlock": testlock,
+                    "name": "Interrupted Thread",
+                },
+            )
+            thread2 = threading.Thread(
+                target=thread_method,
+                kwargs={
+                    "test_instance": self,
+                    "motion_title": "Passing motion",
+                    "meeting_id": 222,
+                    "workflow_id": 13,
+                    "testlock": None,
+                    "name": "Passing Thread",
+                },
+            )
+
+            testlock.acquire()
+            thread1.start()
+            sleep(1)
+            thread2.start()
+            thread2.join()
+            testlock.release()
+            thread1.join()
+
+        self.assert_model_not_exists("motion/1")
+        model2 = self.get_model("motion/2")
+        model3 = self.get_model("motion/3")
+        self.assertNotEqual(model3["sequence_number"], model2["sequence_number"])
```

Comment:
Please be specific. I await that the thread 2 produces 1 and the halted thread 2.

Reply:
This is specific:
Thread-1 gets the next id=1, but has to wait
Thread-2 gets and stores its motion with id=2
Thread-1 continues after acquiring the lock, but fails (id=1 is lost) and retries. On retry it gets the id=3

Reply:
```suggestion
        self.assertEqual(model2["sequence_number"], 2)
        self.assertEqual(model3["sequence_number"], 3)
```

Reply:
Done
except the sequence_numbers are 1 and 2 for the id's 2 and 3 ;-)

Reply:
Ah yes